### PR TITLE
BLADE-252 Use Java stdlib for unzip, and add ZipSlip protection

### PR DESCRIPTION
**Pull request by christopherbryanboyd**

### CODE REVIEW

**Comment by christopherbryanboyd** [comment]
ci:retest

File: cli/src/main/java/com/liferay/blade/cli/util/BladeUtil.java [hunk 1]
```diff
@@ -563,6 +536,54 @@ public static void unzip(File srcFile, File destDir, String entryToStart) throws
 		}
 	}
 
+	public static void unzipStream(InputStream in, File outputDir) throws Exception {
+		unzipStream(in, outputDir, true);
+	}
+
+	public static void unzipStream(InputStream in, File outputDir, boolean preserveNewerFiles) throws Exception {
```
Comment (gamerson) [inline cli/src/main/java/com/liferay/blade/cli/util/BladeUtil.java]: Can you use the zip-slip.zip tests zip that is checked into our repo to make sure that this method will handle it correctly?

File: cli/src/main/java/com/liferay/blade/cli/util/BladeUtil.java [hunk 2]
```diff
@@ -563,6 +536,54 @@ public static void unzip(File srcFile, File destDir, String entryToStart) throws
...
+		try (ZipInputStream stream = new ZipInputStream(in)) {
+			ZipEntry entry;
+
+			Path outputPath = outputDir.toPath();
+
+			while ((entry = stream.getNextEntry()) != null)
+			{
```
Comment (gamerson) [inline cli/src/main/java/com/liferay/blade/cli/util/BladeUtil.java]: `{` never have their own line :)

**Comment by gamerson** [comment]
hey @christopherbryanboyd  please address comments and resend!  Thanks buddy

**Pull request closed without merging**